checker/src/main/java/org/checkerframework/checker/nullness/KeyForAnnotatedTypeFactory.java

```diff
@@ -225,6 +225,64 @@ public boolean isSubtype(AnnotationMirror subAnno, AnnotationMirror superAnno) {
             }
             return super.isSubtype(subAnno, superAnno);
         }
+
+        @Override
+        public AnnotationMirror leastUpperBound(AnnotationMirror a1, AnnotationMirror a2) {
+            if (AnnotationUtils.areSameByName(a1, UNKNOWNKEYFOR)) {
+                return a1;
+            } else if (AnnotationUtils.areSameByName(a2, UNKNOWNKEYFOR)) {
+                return a2;
+            } else if (AnnotationUtils.areSameByName(a1, KEYFORBOTTOM)) {
+                return a2;
+            } else if (AnnotationUtils.areSameByName(a2, KEYFORBOTTOM)) {
+                return a1;
+            } else if (AnnotationUtils.areSameByName(a1, KEYFOR)
+                    && AnnotationUtils.areSameByName(a2, KEYFOR)) {
+                List<String> a1Values = extractValues(a1);
+                List<String> a2Values = extractValues(a2);
+                LinkedHashSet<String> set = new LinkedHashSet<>(a1Values);
+                set.retainAll(a2Values);
+                return createKeyForAnnotationMirrorWithValue(set);
+            }
+            // Ignore annotation values to ensure that annotation is in supertype map.
```

Comment:
At this point, is `a1` or `a2` `PolyKeyFor`? A comment might be helpful. Same in GLB below.